src/services/eventbus.js

```diff
@@ -0,0 +1,31 @@
+import Vue from 'vue'
+import store from '../store'
+
+export const EventBus = new Vue()
+
+function fetchDownload (file) {
```

Comment:
Having the real operation on the event bus seems wrong to me.

The bus is for pure communication. Senders and receivers live outside